### ISSUE TITLE
URL query、parameter 、Header can't load data before it  was created 

### ISSUE THREAD

**Issue by JerryWu1234** [mui/mui-toolpad#2785]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1.
2.
3.

### Current behavior 😯

![image](https://github.com/mui/mui-toolpad/assets/27261522/b9e0ac00-a4a3-4bfe-84dd-e06e3da7e83a)

it can't  load query and parameter that before you was created 

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by Janpot** [comment]
@JerryWu1234 It's hard to follow, can you fill out the template with clear steps on how to reproduce and what is the desired behavior?

**Comment by JerryWu1234** [comment]
first step:
click add-query and open dialog 
![image](https://github.com/mui/mui-toolpad/assets/27261522/406c0749-e1d1-450d-9df4-1b0ce39e4261)
and create `parameter` and `header`
second step 
click save button on the dialog

![image](https://github.com/mui/mui-toolpad/assets/27261522/b319ea7b-de48-4e0f-88ad-c4ddf17099f1)

and all of changes will be in page.yml

and reopen add-query

there is nothing to display

**Comment by Janpot** [comment]
I can't reproduce, this is what I see:


https://github.com/mui/mui-toolpad/assets/2109932/27b5c825-249f-4500-b6a1-1d1adb799110

Does it behave differently for you?
